llama.cpp

```diff
@@ -4072,6 +4080,30 @@ static void llm_load_vocab(
             vocab.special_cls_id  = -1;
             vocab.special_mask_id = -1;
 
+            // For Fill-In-the-Middle (FIM)/infill models which where converted
+            // prior to support of FIM special tokens in GGUF, the following
+            // will allow those models to continue to work. The general names
+            // of the known models are currently CodeLlama (LLM_ARCH_LLAMA) and
+            // CodeGemma (LLM_ARCH_GEMMA). This can potentially be removed once
+            // new versions of these models have been published.
+            std::string gen_name;
+            ml.get_key(LLM_KV_GENERAL_NAME, gen_name);
```

Comment:
Yeah, it's lines 4083 - 4106 that are causing the issue.

Reply:
`examples/train-text-from-scratch/train-text-from-scratch.cpp` doesn't rely on or use `LLM_KV_GENERAL_NAME`, so that's why I'm able to train, but not inference. This most likely has other unintended side-effects due to the implementation.

Reply:
Does https://github.com/ggerganov/llama.cpp/pull/6709 fix the issue?

Reply:
Yeah, I think so. It seems like setting the required parameter to false did the trick.

Reply:
@teleprint-me Sorry about causing this and wasting your time. And thanks @ggerganov for fixing my mistake!